src/client/app/components/admin/users/CreateUserModalComponent.tsx

```diff
@@ -0,0 +1,245 @@
+	ModalBody, ModalFooter, ModalHeader, Row
+} from 'reactstrap';
+import { userApi } from '../../../redux/api/userApi';
+import { User, UserRole, userDefaults } from '../../../types/items';
+import { showErrorNotification, showSuccessNotification } from '../../../utils/notifications';
+import translate from '../../../utils/translate';
+import TooltipHelpComponent from '../../TooltipHelpComponent';
+import TooltipMarkerComponent from '../../TooltipMarkerComponent';
+import { tooltipBaseStyle } from '../../../styles/modalStyle';
+
+/**
+ * Defines the create user modal form
+ * @returns CreateUserModal component
+ */
+export default function CreateUserModal() {
+
+	// create user form state and use the defaults
+	const [userDetails, setUserDetails] = useState(userDefaults);
+
+	// user api
+	const [createUser] = userApi.useCreateUserMutation();
+
+	// check if passwords match
+	useEffect(() => {
+		setUserDetails(prevDetails => ({
+			...prevDetails,
+			passwordMatch: (userDetails.password === userDetails.confirmPassword),
+			passwordLength: userDetails.password.length > 7
+		}));
+	}, [userDetails.password, userDetails.confirmPassword]);
+
+	// check if form is valid
+	const isFormValid = () => {
+		return !!userDetails.username &&
+			!!userDetails.password &&
+			(userDetails.confirmPassword === userDetails.password) &&
+			!!(userDetails.role !== 'invalid');
+	};
+
+	// Handlers for each type of input change
+	const handleStringChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setUserDetails(prevDetails => ({
+			...prevDetails,
+			[e.target.name]: e.target.value
+		}));
+	};
+
+	const handleRoleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setUserDetails(prevDetails => ({
+			...prevDetails,
+			role: e.target.value as UserRole
+		}));
+	};
+
+	// Methods to reset form fields
+	const resetForm = () => {
+		setUserDetails(userDefaults);
+	};
+
+	const resetPasswordFields = () => {
+		setUserDetails({
+			...userDetails, password: '',
+			confirmPassword: ''
+		});
+	};
+	// End of reset form methods
+
+	// Modal show/close
+	const [showModal, setShowModal] = useState(false);
+
+	const handleShowModal = () => setShowModal(true);
+
+	const handleCloseModal = () => {
+		resetPasswordFields();
+		setShowModal(false);
+	};
+	// End Modal show/close
+
+	const handleSubmit = async () => {
+		const newUser: User = { username: userDetails.username, role: userDetails.role, password: userDetails.password, note: userDetails.note };
+		createUser(newUser)
+			.unwrap()
+			.then(() => {
+				showSuccessNotification(translate('users.successfully.create.user') + userDetails.username);
+				resetForm();
+				handleCloseModal();
+			})
+			.catch(error => {
+				showErrorNotification(translate('users.failed.to.create.user') + userDetails.username + ' ' + error.data.message);
+				resetPasswordFields();
+			});
+	};
+
+	// allow enter key to submit form
+	const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
```

Comment:
First, this ability is only on the user pages and not any others. If we are going to keep this then it needs an issue for all pages to do this. Having said that, I have mixed feeling about having enter automatically submit the form. This and other forms have multiple buttons at the bottom so this means OED is picking submit as the default. I could see cancel as an option too. So, I'm leaning toward no default action on enter even though that can be faster. I think it is better to err on not accidentally making changes than make it easier/faster.

Second, if this does stay then the console.log needs to be a popup and translated.

Reply:
I'll take care of it

Reply:
Removed the enter key functionality.